ws/connection/connection.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2020 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package connection
+
+import (
+	"net/http"
+	"net/url"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/gorilla/websocket"
+	"github.com/vmihailenco/msgpack"
+
+	"github.com/mendersoftware/go-lib-micro/ws"
+)
+
+type Connection struct {
+	writeMutex sync.Mutex
+	// the connection handler
+	connection *websocket.Conn
+	// Time allowed to write a message to the peer.
+	writeWait time.Duration
+	// Maximum message size allowed from peer.
+	maxMessageSize int
+	// Time allowed to read the next pong message from the peer.
+	defaultPingWait time.Duration
```

Comment:
These fields are constantly zero (line 68-70). Either remove them or make them variable. Also, a zero value sounds like it's constantly going to fail - unless that means "no limit".

Reply:
fixed, it was only missing the assignment from the arguments.